zulip_bots/zulip_bots/lib.py

```diff
@@ -239,9 +240,14 @@ def extract_query_without_mention(message: Dict[str, Any], client: ExternalBotHa
     the stripped message with the bot's @mention removed.  Otherwise, it returns None.
     """
     mention = '@**' + client.full_name + '**'
-    if not message['content'].startswith(mention):
-        return None
-    return message['content'][len(mention):].lstrip()
+    extended_mention_regex = re.compile(r'^@\*\*.*\|' + str(client.user_id) + r'\*\*')
+    print(extended_mention_regex)
+    extended_mention_match = extended_mention_regex.match(message['content'])
+    if extended_mention_match:
+        return message['content'][extended_mention_match.end():].lstrip()
+    elif message['content'].startswith(mention):
```

Comment:
Stylistically, if an `if` block ends in a `return`, I usually put a blank line after it, and then do an `if`, not an `elif`.  It's not a big deal, I just find it a bit easier to read:

~~~ py
if case_one:
    # bla bla bla
    return foo

if case_two:
    return bar

# We got this far, so we must not have matched on anything
return None
~~~

Reply:
Formatted it accordingly now. I think I initially changed it from that because Tim suggested that he preferred that design pattern. I agree that the early return pattern looks neater, though.

Reply:
I think it previously didn't obviously return in the first code path?  Anyway, looks good now.